### PR TITLE
Fix sdl sending of PerformInteraction unsuccess response to mobile in case choice id selected via VR

### PR DESCRIPTION
**Pull request by YarikMamykin**
ATF Test Scripts to check [#2830](https://github.com/smartdevicelink/sdl_core/issues/2830) & [#3112](https://github.com/smartdevicelink/sdl_core/issues/3112)

This PR is **ready** for review.

### Summary
Added cases for PerformInteraction response handling in VR only/BOTH modes
Updated existing tests.

### ATF version
develop

### CLA
- [X] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: test_scripts/Smoke/API/039_PerfomInteraction_Non_Media_PositiveCase_SUCCESS.lua [hunk 1]
```diff
@@ -127,11 +127,11 @@ local function SendOnSystemContext(self, ctx)
     { appID = commonSmoke.getHMIAppId(), systemContext = ctx })
 end
 
---! @setExChoiseSet: ChoiceSet structure for UI.PerformInteraction request
+--! @setExChoiceSet: ChoiceSet structure for UI.PerformInteraction request
```
Comment (aderiabin) [inline test_scripts/Smoke/API/039_PerfomInteraction_Non_Media_PositiveCase_SUCCESS.lua]: @GetmanetsIrina Please update function name of function `setChoiseSet()` too. (it has the same missprint)

File: test_scripts/Smoke/API/012_PerfomInteraction_PositiveCase_SUCCESS.lua [hunk 2]
```diff
@@ -385,6 +385,114 @@ local function PI_PerformViaBOTH(paramsSend, self)
...
+        self.hmiConnection:SendNotification("TTS.Stopped")
+        self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS",
+          { choiceID = paramsSend.interactionChoiceSetIDList[1] })
+        SendOnSystemContext(self,"MAIN")
+      end
+      RUN_AFTER(uiResponse, 3000)
+    end)
+  ExpectOnHMIStatusWithAudioStateChanged_PI(self, "BOTH")
+  self.mobileSession1:ExpectResponse(cid, { success = true, resultCode = "SUCCESS",
+  choiceID = paramsSend.interactionChoiceSetIDList[1], triggerSource = "MENU" })
```
Comment (dboltovskyi) [inline test_scripts/Smoke/API/012_PerfomInteraction_PositiveCase_SUCCESS.lua]: @GetmanetsIrina Please add indentation here and in other such places

File: test_scripts/Smoke/API/012_PerfomInteraction_PositiveCase_SUCCESS.lua [hunk 3]
```diff
@@ -400,8 +508,11 @@ runner.Step("CreateInteractionChoiceSet no VR commands with id 400", CreateInter
 runner.Title("Test")
 runner.Step("PerformInteraction with VR_ONLY interaction mode", PI_PerformViaVR_ONLY, {requestParams})
```
Comment (dboltovskyi) [inline test_scripts/Smoke/API/012_PerfomInteraction_PositiveCase_SUCCESS.lua]: @GetmanetsIrina Please update `PI_PerformViaVR_ONLY` function accordingly. Now it contains SUCCESS UI response.

**Comment by GetmanetsIrina** [comment]
@aderiabin, @dboltovskyi, updated according to comments in 
71c7a4e . Please review.

**Comment by YarikMamykin** [comment]
@theresalech and @jordynmackool this PR passed internal Luxoft review and ready for Livio team review. Please, take a look.

**Comment by jordynmackool** [comment]
@YarikMamykin thanks, we will review and let you know if we have any questions.

**Comment by GetmanetsIrina** [comment]
@iCollin , branch fix/sdl_sends_unsuccess_PI_in_case_choiceID_selected_via_VR is rebased on actual develop.

**Pull request merged**